### ISSUE TITLE
Why is @Override required for the @CallSuper annotation to be enforced?

### ISSUE THREAD

**Issue by mstevens83** [kotcrab/callsuper-annotation#1]
Great project! But I do have one comment...

I agree that, in general, it is a good idea to annotate any overriding method with `@Override`. But was it really necessary to make the enforcing of the  `@CallSuper` annotation depend on the overriding method having the `@Override` annotation?
What if I'm so sloppy that I forget both the `@Override` annotation AND the super call? Currently I wouldn't not get any error or warning in such cases, which seems less than ideal.

So if possible I'd remove the dependency on there being an `@Override` annotation. After all you already provide a more explicit mechanism to disable the `@CallSuper` check, namely the `@OverrideCallSuper` annotation.

Thanks for taking this into consideration. If you think it is possible but don't have the time I can have a try myself, but I've never played with implementing annotations so it might take me a while to figure it out.

**Comment by mstevens83** [comment]
Looking at the code it actually seems really simple. All that needs changing is this line: https://github.com/kotcrab/callsuper-annotation/blob/master/src/com/kotcrab/annotation/callsuper/CallSuperProcessor.java#L45

However, looking at it I'm beginning to realise why you did it. I'm guessing it's for performance, right? By assuming no-one forgets `@Override` (or IDEs add it), you can limit the number of methods for which the check must run. Is that it?

**Comment by kotcrab** [comment]
The annotation processor is triggered by `@Override` annotation. If you change something in method with `@Override` it will trigger processor that checks if super call is done, IIRC it wasn't possible to do it other way.

It's not ideal but you can set in your IDE to show build errors when you have missing `@Override`

**Comment by kotcrab** [comment]
Also if you want to look into it take note that making and debugging annotation processors is hard. You can use this snippet to run annotation processor with debugger support:

``` java
public class Main {
    public static void main (String[] args) throws IOException {
        File searchDir = new File("E:\\test-project\\src");

        JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
        StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
        fileManager.setLocation(StandardLocation.SOURCE_PATH, singleton(searchDir.getAbsoluteFile()));
        Iterable<JavaFileObject> sources = fileManager.list(StandardLocation.SOURCE_PATH, "", singleton(JavaFileObject.Kind.SOURCE), true);

        JavaCompiler.CompilationTask task = compiler.getTask(null, fileManager, null, null, null, sources);
        task.setProcessors(singleton(new CallSuperProcessor()));
        task.call();
        System.out.println("Done");
    }
}
```

**Comment by mstevens83** [comment]
I see, I was guessing there would be a way to get all methods from the environment, instead of only those with a certain annotation.

**Comment by mstevens83** [comment]
I've spent a couple of hours digging into this but so far found no way to make things work without `@Override`. The problem is that contrary to class-level annotation there is no inheritance of method-level annotations in Java. But I guess you knew that already.
Anyway, always nice to learn a thing or two.
Feel free to close this issue if you want. I'll reopen it should I even find a way to make it work without `@Override`.

**Comment by kotcrab** [comment]
Yeah, annotation processing wasn't meant to be used that way, there isn't even option to access method source without using proprietary Sun Trees API. I doubt it's possible to do it with standard annotation processors, maybe better way would be to write plugins for IDEs and build tools but that's something I don't have time to do.

**Comment by vovcacik** [comment]
Yeah, the algorithm is bottom up, so kotcrab's processor is not listening for `@CallSuper`, but for `@Override`. There is no simple/obvious way to rewrite it in top down fashion, e.g. you can't get list of overriding methods for a given method.

Can you please list this known issue on homepage?
